### DISCUSSION TITLE
Adding only 90 degrees rotations to augmentations

### DISCUSSION THREAD

**Discussion by ivankovd** [ultralytics/yolov5#5994] [Q&A] [answered]
Hi! I wonder what exactly should I pass to 'degrees' param in hyp.scratch.yaml in order to add only [0, 90, 180, 270] degrees rotations to augmentations during network training? I've found several same or related questions in issues but no direct answers were provided.
As I understood from the documentation, 'degrees' param could be 0 or +-180 (btw does the sign mean that if I will put here '+180' - only clockwise rotations will be applied and if '-180' - only counterclockwise?), so what should I do if I want both clock and counterclockwise rotations range?
Also, I found [commented out line of code](https://github.com/ultralytics/yolov5/blob/da9a1b719ba7d10e209ff89efe28b074fb9a5f16/utils/augmentations.py#L145) in random_perspective function in https://github.com/ultralytics/yolov5/blob/master/utils/augmentations.py which looks like what I need, but I suppose there should be more simple way to make it using the hyperparameters config mentioned earlier.
Thanks anybody in advance for help!

**Comment by glenn-jocher** [comment, marked as answer]
@ivankovd you can simply uncomment L145 here to allow discrete random 90deg rotations:
https://github.com/ultralytics/yolov5/blob/c1249a47c7fe19e2067cb25ed8347e67d26ff1f1/utils/augmentations.py#L142-L149
